src/slider/internal.tsx

```diff
@@ -0,0 +1,188 @@
+  }
+
+  const tooltip = hasValue && <Tooltip value={valueFormatter ? valueFormatter(value) : value} trackRef={handleRef} />;
+
+  const getAriaValueText = () => {
+    if (value !== undefined && valueFormatter && valueFormatter(value)) {
+      return valueFormatter(value);
+    }
+
+    if (value && referenceValues && valueFormatter && !valueFormatter(value)) {
+      const valueArray = [min, ...referenceValues, value, max];
+      const prevAndNext = findLowerAndHigherValues(valueArray, value);
+      const prev = prevAndNext.lower && valueFormatter(prevAndNext.lower);
+      const next = prevAndNext.higher && valueFormatter(prevAndNext.higher);
+
+      if (prev === undefined) {
+        return `${value}, between ${valueFormatter(min)} and ${next}`;
+      }
+      if (next === undefined) {
+        return `${value}, between ${prev} and ${valueFormatter(max)}`;
+      }
+
+      return `${value}, between ${prev} and ${next}`;
+    }
+
+    return undefined;
+  };
+
+  return (
+    <div {...baseProps} ref={__internalRootRef} className={clsx(baseProps.className, styles.root)}>
+      <div className={styles.slider}>
+        <div
+          ref={handleRef}
+          className={clsx(styles['tooltip-thumb'])}
+          style={{
+            [customCssProps.sliderTooltipPosition]: `calc(${percent}% - ${tooltipPositionAdjust}px)`,
+          }}
+        />
+        {showTooltip && tooltip}
+        <div
+          className={clsx(styles['slider-track'], {
+            [styles.disabled]: disabled,
+          })}
+        />
+        {step && tickMarks && (
+          <div
+            className={clsx(styles.ticks)}
+            style={{
+              [customCssProps.sliderTickCount]: Math.round((max - min) / step),
+            }}
+          >
+            {getStepArray(step, [min, max]).map((step, index) => (
+              <div
+                key={`step-${index}`}
+                className={clsx(styles.tick, {
+                  [styles.filled]: !hideFillLine && hasValue && value > step,
+                  [styles.error]: invalid && !hideFillLine && hasValue && value > step,
+                  [styles['error-active']]: invalid && isActive && !hideFillLine && hasValue && value > step,
+                  [styles.disabled]: disabled,
+                })}
+              ></div>
+            ))}
+          </div>
+        )}
+        {!hideFillLine && (
+          <div
+            className={clsx(styles['slider-range'], {
+              [styles.error]: invalid,
+              [styles.active]: isActive,
+              [styles['error-active']]: invalid && isActive,
+              [styles.disabled]: disabled,
+            })}
+            style={{ [customCssProps.sliderRangeInlineSize]: `${percent}%` }}
+          />
+        )}
+      </div>
+
+      <input
+        aria-label={ariaLabel}
+        aria-labelledby={ariaLabel && !rest.ariaLabelledby ? undefined : ariaLabelledby}
```

Comment:
why do we check for `!rest.ariaLabelledby` here?

Reply:
Same behavior as Input. I'll add a comment.
```
    // aria-labelledby has precedence over aria-label in accessible name calculation.
    // When aria-label is provided for Input, it should override aria-labelledBy from form-field context.
    // If both aria-label and aria-labelledby come from Input props, aria-labelledby will be used in accessible name
```